graylog2-server/src/main/java/org/graylog2/migrations/MigrationHelpers.java

```diff
@@ -86,10 +86,18 @@ public String ensureBuiltinRole(String roleName, String description, Set<String>
     @Nullable
     public String ensureUser(String userName, String password, String firstName, String lastName, String email,
                              Set<String> expectedRoles) {
+        return ensureUser(userName, password, firstName, lastName, email, expectedRoles, false);
+    }
+
+    @Nullable
+    public String ensureUser(String userName, String password, String firstName, String lastName, String email,
+                             Set<String> expectedRoles, boolean isServiceAccount) {
         User previousUser = null;
         try {
             previousUser = userService.load(userName);
-            if (previousUser == null || !previousUser.getRoleIds().containsAll(expectedRoles)) {
+            if (previousUser == null
+                    || !previousUser.getRoleIds().containsAll(expectedRoles)
+                    || (isServiceAccount && !previousUser.isServiceAccount())) {
```

Comment:
Hmm, I think we need to add a check for the opposite case as well. `isServiceAccount` is `false` but `previousUser.isServiceAccount` is true. I guess the following should work?

```suggestion
                    || !Objects.equals(isServiceAccount, previousUser.isServiceAccount())) {
```

Reply:
Oops - missed that. I don't think this is a scenario that can happen with the existing calls to ensureUser(). But it is what you'd expect ensureUser() to do.